### DISCUSSION TITLE
FastAPI Documentation update regarding Pydantic usage

### DISCUSSION THREAD

**Discussion by meer-khan** [fastapi/fastapi#11031] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/book", status_code=status.HTTP_201_CREATED, response_model = Post)
def create_book():
    ...


class Book(BaseModel):
    title: str 
    content: str
```

### Description

I hope this message finds you well. I've been using FastAPI and noticed a potential need for an update in the documentation regarding Pydantic usage. Specifically, with the latest version of Pydantic (2.0), the orm_mode attribute in the Config class of a Pydantic model is no longer required when setting the response_model argument.

In the official FastAPI documentation under the section https://fastapi.tiangolo.com/tutorial/sql-databases/#use-pydantics-orm_mode,  it still mentions the use of orm_mode = True. However, as of Pydantic 2.0, this is no longer necessary.

I wanted to bring this to your attention and inquire if there are plans to update the documentation. If needed, I am more than willing to contribute to this update. Please let me know your thoughts on this matter. 

![image](https://github.com/tiangolo/fastapi/assets/40295656/b608e44b-5700-41fa-b55b-7102c7b17b2a)

### Operating System

Windows

### FastAPI Version

0.109.0

### Pydantic Version

2.5.3

### Python Version

3.11.0

**Comment by Kludex** [comment]
The docs are in a state that supports both Pydantic v1 and v2.

**Comment by meer-khan** [comment]
Thank you for your swift response, Kludex. Indeed, I acknowledge that the documentation is designed to accommodate both versions of Pydantic. However, considering that the latest version of FastAPI defaults to downloading Pydantic 2.5.3, it would be beneficial to align the documentation with the latest functionalities introduced in Pydantic. For instances where an older version of FastAPI utilizes Pydantic v1, the existing documentation can be referenced. It's worth noting that Pydantic is on the verge of releasing version 3, rendering functionalities of v1 obsolete. Hence, updating the FastAPI documentation to reflect the Pydantic version bundled with the current FastAPI installation would be a prudent approach. What do you think about it?

**Reply by Kludex** [reply]
My previous answer was actually an answer that @tiangolo gave before in another GitHub discussion.

**Reply by meer-khan** [reply]
Yeah, that's completely understandable that this was the exact response of @tiangolo in some other discussion, but do you have your own thoughts on my suggestion? FastAPI is directly dependent on the Pydantic, that is why I am raising this concern for the better experience in official docs and to make open source tools and technologies more stronger.

**Reply by meer-khan** [reply]
@tiangolo.- 

 Hi @Kludex, I am very disappointed by your unprofessional response. Brother at least you should appropriately answer me rather than reacting with a "thumbs down".  

This is a discussion forum, not a GitHub issue, and we can discuss here anything we want, it's not like I am degrading someone or something, I am just helping FOSS to be more robust with changes, if you think that my suggestion or point of discussion is not legit/genuine, then you can answer me with a good logic. we all are programmers and we need logical answers, I do not find any logic in the answer **"My answer was actually an answer that @tiangolo gave before in another GitHub discussion."** it is like you do not have your own thoughts or views on my suggestion and you are just copy-pasting the answers from author, and you are marking your comment as an answer even if it's not a legit answer at all. 

Hope you will understand my point. Thank you

**Comment by estyrke** [comment, marked as answer]
I agree with the opinion that the documentation should reflect the current version, but see the note section on the top of that page:

> These docs are about to be updated. 🎉
>
> The current version assumes Pydantic v1, and SQLAlchemy versions less than 2.0.
>
> The new docs will include Pydantic v2 and will use [SQLModel](https://sqlmodel.tiangolo.com/) (which is also based on SQLAlchemy) once it is updated to use Pydantic v2 as well.

I think this makes it clear that the authors are aware of the issue and have an update in the pipeline. Just have patience. 🙂

**Reply by meer-khan** [reply]
@estyrke Thank you for your response buddy, yes this is what I was looking for, something reasonable. Thanks a lot, and Thankyou Kludex. I apologize if I sounded too straightforward. FOSS is the best.
